src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/withToolbar.js

```diff
@@ -0,0 +1,25 @@
+// @flow
+import {autorun} from 'mobx';
+import toolbarStore from './stores/ToolbarStore';
+
+export default function withToolbar(Component) {
+    return class WithToolbar extends Component {
+        componentWillMount() {
+            if (super.componentWillMount) {
+                super.componentWillMount();
+            }
+
+            this.disposer = autorun(() => {
+                toolbarStore.setItems(this.getToolbar());
```

Comment:
The call to `getToolbar` is the reason I have to use inheritance here. It is an instance method, and cannot be called if I just decorate it. I would like to use return a new component without inheritance here, but I haven't found a way to make this work...

The other way I have found adds a bit more code and more overhead... There the other component would not be decorated, but we would add a provider in between:

```javascript
class Form extends React.Component {
    getToolbar() { /* ... */ }

    render() {
        return (
            <ToolbarProvider toolbar={this.getToolbar()}>
                <OtherComponents />
            </ToolbarProvider>
        );
    }
}
```

In this scenario the `ToolbarProvider` would only set the items on the `ToolbarStore`. There would be no need for `autorun` at all, but I think it would take React a little bit more time to determine what should be rendered again (I have played a bit with that solution as well, and realized that at least the `render` method of `OtherComponents` is called every time... I am wondering if that is something to worry about, or if the solution with the provider is still the better one...